flax/core/frozen_dict.py

```diff
@@ -61,13 +61,20 @@ def __hash__(self):
       self._hash = h
     return self._hash
 
-  def copy(self, **add_or_replace):
-    return type(self)(self, **add_or_replace)
+  def copy(self, add_or_replace):
+    return type(self)(self, **unfreeze(add_or_replace))
 
   def items(self):
     for key in self._dict:
       yield (key, self[key])
 
+  def pop(self, key):
```

Comment:
the reason for using the same signature is because this is essentially the functional version of `pop`. Although perhaps we might want to add the option for adding a `default` arg.